src/otx/core/model/detection_3d.py

```diff
@@ -295,3 +216,349 @@ def get_classification_layers(self, prefix: str = "model.") -> dict[str, dict[st
                 num_extra_classes = 6 * sample_model_dim - 5 * incremental_model_dim
                 classification_layers[prefix + key] = {"stride": stride, "num_extra_classes": num_extra_classes}
         return classification_layers
+
+
+class MonoDETRModel(ImageModel):
+    """A wrapper for MonoDETR 3d object detection model."""
+
+    __model__ = "mono_3d_det"
+
+    def __init__(self, inference_adapter: InferenceAdapter, configuration: dict[str, Any], preload: bool = False):
+        """Initializes a 3d detection model.
+
+        Args:
+            inference_adapter (InferenceAdapter): inference adapter containing the underlying model.
+            configuration (dict, optional): configuration overrides the model parameters (see parameters() method).
+              Defaults to dict().
+            preload (bool, optional): forces inference adapter to load the model. Defaults to False.
+        """
+        super().__init__(inference_adapter, configuration, preload)
+        self._check_io_number(3, 5)
+
+    def preprocess(self, inputs: dict[str, np.ndarray]) -> tuple[dict[str, Any], ...]:
+        """Preprocesses the input data for the model.
+
+        Args:
+            inputs (dict[str, np.ndarray]): a dict with image, calibration matrix, and image size
```

Comment:
How about adding explanation of inputs with more details? It looks a little bit hard to know which values are necessary in `inputs` with current docstring.

Reply:
That makes no sense at the moment, since the exact input shape is not yet defined. We can't leave it like it is now